src/app/clients/add/helperFunctions.tsx

```diff
@@ -0,0 +1,286 @@
+import React from "react";
+import { SelectChangeEvent } from "@mui/material";
+import { booleanGroup } from "@/components/DataInput/inputHandlerFactories";
+import supabase, { InsertSchema } from "@/supabase";
+import { Database } from "@/database_types_file";
+
+type Event = React.ChangeEvent<HTMLInputElement> | SelectChangeEvent;
+type OnChange = (event: Event) => void;
+type OnChangeCheckbox = (event: React.ChangeEvent<HTMLInputElement>) => void;
+type ErrorSetter = (errorKey: string, errorType: Error) => void;
+type FieldSetter = (fieldKey: string, newFieldValue: Field) => void;
+type Field = string | number | (boolean | null) | booleanGroup | Address | Person[];
+type PersonType = Database["public"]["Enums"]["gender"];
+type FamilyDatabaseRecord = InsertSchema["families"];
+export type ClientDatabaseRecord = InsertSchema["clients"];
+export enum Error {
+    initial = "required",
+    none = "",
+    required = "This is a required field.",
+    invalid = "Please enter a valid entry.",
+    submit = "Please ensure all fields have been entered correctly. Required fields are labelled with an asterisk.",
+    database = "An error has occurred. Please try again later.",
+}
+
+export const phoneNumberRegex = /^([+]?[(]?[0-9]{3}[)]?[-\s.]?[0-9]{3}[-\s.]?[0-9]{4,6})?$/;
+// Regex source: https://ihateregex.io/expr/phone/
+
+export const postcodeRegex =
+    /^([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})$/;
+// Regex source: https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/488478/Bulk_Data_Transfer_-_additional_validation_valid_from_12_November_2015.pdf
+
+export const numberRegex = /^\d+$/;
+
+interface Address {
+    line1: string;
+    line2: string;
+    town: string;
+    county: string;
+    postcode: string;
+}
+
+export interface Person {
+    personType: PersonType;
+    age?: number | null;
+    quantity?: number;
+}
+
+export interface ErrorType {
```

Comment:
Errors are looking significantly better. Is there a better name for this interface along the lines of `FormErrors` or similar?